x/incentives/keeper/distribute.go

```diff
@@ -344,32 +373,80 @@ func (k Keeper) Distribute(ctx sdk.Context, gauges []types.Gauge) (sdk.Coins, er
+		// get pool Id corresponding to the gaugeId
+		incParams := k.GetParams(ctx).DistrEpochIdentifier
+		currEpoch := k.ek.GetEpochInfo(ctx, incParams)
+
+		isCLPoolGauge, pool := k.IsValidConcentratedLiquidityGauge(ctx, gauge.Id, currEpoch.Duration)
+		// only want to run this logic if the gaugeId is associated with CL PoolId
+		if isCLPoolGauge {
+			for _, coin := range gauge.Coins {
+				// since we're only doing osmo incentive we can assume there to be only 1 coin therefore coins[0]
+				incentiveRecord := cltypes.IncentiveRecord{
+					PoolId:           pool.GetId(),
+					IncentiveDenom:   coin.Denom,
+					IncentiveCreator: k.ak.GetModuleAddress(types.ModuleName),
+					RemainingAmount:  coin.Amount.ToDec(),
+					// calculate amount of tokens to emit per second
+					// for ex: 10000tokens to be distributed over 1day epoch will be 1000 tokens ÷ 86,400 seconds ≈ 0.01157 tokens per second (truncated)
+					EmissionRate: sdk.NewDecFromInt(coin.Amount).QuoTruncate(sdk.NewDec(int64(currEpoch.Duration.Seconds()))),
+					StartTime:    gauge.GetStartTime(),
+					MinUptime:    currEpoch.Duration,
+				}
+
+				coinsToDistribute, err := k.distributeConcentratedLiquidityInternal(ctx, incentiveRecord, gauge)
+				if err != nil {
+					return nil, err
+				}
+
+				gaugeDistributedCoinsCL = coinsToDistribute
+			}
 		} else {
-			gaugeDistributedCoins, err = k.distributeInternal(ctx, gauge, filteredLocks, &distrInfo)
-		}
-		if err != nil {
-			return nil, err
+			filteredLocks := k.getDistributeToBaseLocks(ctx, gauge, locksByDenomCache)
+			// send based on synthetic lockup coins if it's distributing to synthetic lockups
+			var err error
+			if lockuptypes.IsSyntheticDenom(gauge.DistributeTo.Denom) {
+				gaugeDistributedCoins, err = k.distributeSyntheticInternal(ctx, gauge, filteredLocks, &distrInfo)
+			} else {
+				gaugeDistributedCoins, err = k.distributeInternal(ctx, gauge, filteredLocks, &distrInfo)
+			}
+			if err != nil {
+				return nil, err
+			}
 		}
-		totalDistributedCoins = totalDistributedCoins.Add(gaugeDistributedCoins...)
+
+		totalDistributedCoins = totalDistributedCoins.Add(gaugeDistributedCoins...).Add(gaugeDistributedCoinsCL...)
 	}
 
 	err := k.doDistributionSends(ctx, &distrInfo)
 	if err != nil {
 		return nil, err
 	}
+
 	k.hooks.AfterEpochDistribution(ctx)
 
 	k.checkFinishDistribution(ctx, gauges)
 	return totalDistributedCoins, nil
 }
 
+// IsValidConcentratedLiquidityGauge checks if a specific gaugeId is is associated with a poolId. It also checks if the poolType is CL pool.
+// If yes return true, otherwise false.
+func (k Keeper) IsValidConcentratedLiquidityGauge(ctx sdk.Context, gaugeId uint64, duration time.Duration) (bool, poolmanagertypes.PoolI) {
+	poolId, err := k.pik.GetPoolIdFromGaugeId(ctx, gaugeId, duration)
+	if err != nil {
+		return false, nil
+	}
+
+	pool, err := k.pmk.GetPool(ctx, poolId)
+	if err != nil {
+		return false, nil
+	}
+
+	return pool.GetType() == poolmanagertypes.Concentrated, pool
```

Comment:
The second return of `pool` does not fit the abstraction of `IsValidConcentratedLiquidityGauge`

Consider getting the pool from the caller, passing in the `poolType` as a parameter and returning a single boolean

Reply:
i actually changed it to `GetValidConcentratedLiquidityGauge` and return Pool, lmk if that makes sense